src/common/blkdev.h

```diff
@@ -17,10 +21,16 @@ enum blkdev_prop_t {
 extern int get_device_by_path(const char *path, char* partition, char* device, size_t max);
 
 
-extern std::string get_device_id(const std::string& devname);
+extern std::string get_device_id(const std::string& devname,
+				 std::string *err=0);
 extern void get_dm_parents(const std::string& dev, std::set<std::string> *ls);
-extern int block_device_run_smartctl(const char *device, int timeout,
-				     std::string *result);
+extern int block_device_get_metrics(const string& devname, int timeout,
```

Comment:
@liewegas Compile failed. This line should change "string&" to "std::string&"
extern int block_device_get_metrics(const std::string& devname, int timeout,

Reply:
@hsiang41 See https://github.com/ceph/ceph/pull/25783